"use strict";exports.id=984,exports.ids=[984],exports.modules={3242:(e,t,s)=>{s.d(t,{c:()=>useAsyncOperation,h:()=>useApi});var a=s(6689);let useApi=(e,t=[],s={})=>{let{throwOnError:r=!1,fallbackData:u=null}=s,[l,n]=(0,a.useState)(u),[c,o]=(0,a.useState)(!0),[i,d]=(0,a.useState)(null),h=(0,a.useCallback)(async(...t)=>{try{o(!0),d(null);let s=await e(...t),a=s?.data!==void 0?s.data:s;return n(a),a}catch(t){let e=t.response?.data?.message||t.message||"An error occurred";if(d(e),t.response?.status===404||!r)return n(u),u;throw t}finally{o(!1)}},t);(0,a.useEffect)(()=>{h()},t);let p=(0,a.useCallback)(()=>h(),[h]);return{data:l,loading:c,error:i,execute:h,refetch:p}},useAsyncOperation=()=>{let[e,t]=(0,a.useState)(!1),[s,r]=(0,a.useState)(null),u=(0,a.useCallback)(async e=>{try{t(!0),r(null);let s=await e();return s}catch(t){let e=t.response?.data?.message||t.message||"An error occurred";throw r(e),t}finally{t(!1)}},[]),l=(0,a.useCallback)(()=>r(null),[]);return{loading:e,error:s,execute:u,clearError:l}}},6761:(e,t,s)=>{s.a(e,async(e,a)=>{try{s.d(t,{a:()=>useAuth});var r=s(6689),u=s(8786),l=e([u]);u=(l.then?(await l)():l)[0];let useAuth=()=>{let e=(0,r.useContext)(u.ZP);if(!e)throw Error("useAuth must be used within an AuthProvider");return e};a()}catch(e){a(e)}})}};